tests/unit/test_services.py

```diff
@@ -696,3 +689,338 @@ def test_fix_code_doesnt_remove_variable(
+                      - item
+                      # Comment: desired: 0 new lines before this comment
+                      - item
+
+                    key: value
+                    """
+                ),
+                None,
+                dedent(
+                    """\
+                    ---
+                    list:
+                      - item
+                      # Comment: desired: 0 new lines before this comment
+                      - item
+                    key: value
+                    """
+                ),
+            ),
+            (
+                dedent(
+                    """\
+                    ---
+                    list:
+                      - item
+                      # Comment: desired: 0 new lines before this comment
+                      - item
+
+                    key: value
+                    """
+                ),
+                YamlfixConfig(comments_optional_number_whitelines_from_content=0),
+                dedent(
+                    """\
+                    ---
+                    list:
+                      - item
+                      # Comment: desired: 0 new lines before this comment
+                      - item
+                    key: value
+                    """
+                ),
+            ),
+            (
+                dedent(
+                    """\
+                    ---
+                    list:
+                      - item
+                      # Comment: desired: 0 new lines before this comment
+                      - item
+
+                    key: value
+                    """
+                ),
+                YamlfixConfig(comments_optional_number_whitelines_from_content=2),
+                dedent(
+                    """\
+                    ---
+                    list:
+                      - item
+                      # Comment: desired: 0 new lines before this comment
+                      - item
+                    key: value
+                    """
+                ),
+            ),
+            (
+                dedent(
+                    """\
+                    ---
+                    list:
+                      - item
+
+                      # Comment: desired: 0 new lines before this comment
+                      - item
+
+                    key: value
+                    """
+                ),
+                None,
+                dedent(
+                    """\
+                    ---
+                    list:
+                      - item
+                      # Comment: desired: 0 new lines before this comment
+                      - item
+                    key: value
+                    """
+                ),
+            ),
+            (
+                dedent(
+                    """\
+                    ---
+                    list:
+                      - item
+
+                      # Comment: desired: 0 new lines before this comment
```

Comment:
Now that I see this test, I can imagine some examples where I would like to have whitelines in list elements, for example an ansible playbook. Would it be difficult to implement?

Reply:
Will evaluate the effort tomorrow or Saturday, and I will circle back

Reply:
It would be possible to implement this like so:

1. Replace [this regex](https://github.com/lyz-code/yamlfix/blob/98649b4f56db5e3521d5e4c9f696052e815e581a/src/yamlfix/adapters.py#L630) by `"\n\n+[\t ]{0,}[#]"`
1.1 The expression `[\t ]{0,}` matches `tab` or `space` between `0` and `unlimited` number of times
2. Instead of using a fixed string [here](https://github.com/lyz-code/yamlfix/blob/98649b4f56db5e3521d5e4c9f696052e815e581a/src/yamlfix/adapters.py#L633-L637) (i.e. `desired_whitelines_with_comments`), use a method that:
2.1 Replaces all of the `\n` by the desired number of `\n` 
2.2 and keep all of the trailing `\t`, `space` and the final character (`#`)


If it's okay with you, I'd vote for opening a new issue and implementing it in a different PR.
I can do the implementation.

Reply:
It's an enhancement over your actual proposal. I'm fine with opening a new issue for this

Reply:
https://github.com/lyz-code/yamlfix/issues/210